src/main/java/Decode.kt

```diff
@@ -0,0 +1,45 @@
+object Decode {
+    private val CHARSET_REV = byteArrayOf( // The Bech32 character set for decoding.
+        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+        -1, -1, -1, -1, -1, -1, 15, -1, 10, 17, 21, 20, 26, 30, 7, 5, -1, -1, -1, -1, -1,
+        -1, -1, 29, -1, 24, 13, 25, 9, 8, 23, -1, 18, 22, 31, 27, 19, -1, 1, 0, 3, 16, 11,
+        28, 12, 14, 6, 4, 2, -1, -1, -1, -1, -1, -1, 29, -1, 24, 13, 25, 9, 8, 23, -1, 18,
+        22, 31, 27, 19, -1, 1, 0, 3, 16, 11, 28, 12, 14, 6, 4, 2, -1, -1, -1, -1, -1
+    )
+
+    fun decodingString(stringToDecode: String): BechData {
+        var lower = false
+        var upper = false
+        if (stringToDecode.length < 8)
+            throw Exception("Input too short: " + stringToDecode.length)
+        if (stringToDecode.length > 90)
+            throw Exception("Input too long: " + stringToDecode.length)
+        for (element in stringToDecode) {
+            if (element.code < 33 || element.code > 126) throw Exception ("Character is invalid")
```

Comment:
Nitpick: I find the way you check if the string isn't mixed case needlessly complicated (and long)